tls/s2n_resume.c

```diff
@@ -799,8 +799,8 @@ int s2n_encrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *
 
     /* Ensure we never encrypt with a zero-filled key */
     uint8_t zero_block[S2N_AES256_KEY_LEN] = { 0 };
-    POSIX_ENSURE(memcmp(key->aes_key, zero_block, S2N_AES256_KEY_LEN) != 0, S2N_ERR_KEY_CHECK);
-
+    POSIX_ENSURE(s2n_constant_time_equals(key->aes_key, zero_block, S2N_AES256_KEY_LEN) != 1,
+            S2N_ERR_KEY_CHECK);
```

Comment:
s2n_constant_time_equals returns a bool, not an int: https://github.com/aws/s2n-tls/blob/ac5878b24f7082ed0397cfd7aaef9c8c3eb33522/utils/s2n_safety.h#L48
```suggestion
    POSIX_ENSURE(!s2n_constant_time_equals(key->aes_key, zero_block, S2N_AES256_KEY_LEN),
            S2N_ERR_KEY_CHECK);
```

Reply:
I see that the comment is wrong though. Someone probably changed the signature + implementation at some point without updating the comment.

Reply:
https://github.com/aws/s2n-tls/pull/2072